src/HeapDump/GCHeapDumper.cs

```diff
@@ -296,9 +297,113 @@ private CollectionMetadata DumpHeapFromProcessDump(string processDumpFile)
+        {
+            return dataReader.EnumerateModules();
+        }
+
+        public void Flush()
+        {
+            dataReader.Flush();
+        }
+
+        public Microsoft.Diagnostics.Runtime.Architecture GetArchitecture()
+        {
+            return dataReader.GetArchitecture();
+        }
+
+        public uint GetPointerSize()
+        {
+            return dataReader.GetPointerSize();
+        }
+
+        public bool GetThreadContext(uint threadID, uint contextFlags, uint contextSize, IntPtr context)
+        {
+            return dataReader.GetThreadContext(threadID, contextFlags, contextSize, context);
+        }
+
+        public bool GetThreadContext(uint threadID, uint contextFlags, uint contextSize, byte[] context)
+        {
+            return dataReader.GetThreadContext(threadID, contextFlags, contextSize, context);
+        }
+
+        public ulong GetThreadTeb(uint thread)
+        {
+            return dataReader.GetThreadTeb(thread);
+        }
+
+        public void GetVersionInfo(ulong baseAddress, out VersionInfo version)
+        {
+            dataReader.GetVersionInfo(baseAddress, out version);
+        }
+
+        public uint ReadDwordUnsafe(ulong addr)
+        {
+            return dataReader.ReadDwordUnsafe(SignExtend(addr));
+        }
+
+        public bool ReadMemory(ulong address, byte[] buffer, int bytesRequested, out int bytesRead)
+        {
+            return dataReader.ReadMemory(SignExtend(address), buffer, bytesRequested, out bytesRead);
+        }
+
+        public bool ReadMemory(ulong address, IntPtr buffer, int bytesRequested, out int bytesRead)
+        {
+            return dataReader.ReadMemory(SignExtend(address), buffer, bytesRequested, out bytesRead);
+        }
+
+        public ulong ReadPointerUnsafe(ulong addr)
+        {
+            return dataReader.ReadPointerUnsafe(SignExtend(addr));
+        }
+
+        public bool VirtualQuery(ulong addr, out VirtualQueryData vq)
+        {
+            return dataReader.VirtualQuery(addr, out vq);
+        }
+
+        private static ulong SignExtend(ulong address)
+        {
+            unsafe
+            {
+                return unchecked((ulong)(long)(IntPtr)(void*)address);
+            }
+        }
+    }
+
     private void InitializeClrRuntime(string processDumpFile, out DataTarget target, out ClrRuntime runtime)
     {
-        target = DataTarget.LoadCrashDump(processDumpFile);
+        var reader = (IDataReader)Activator.CreateInstance(typeof(DataTarget).Assembly.GetType("Microsoft.Diagnostics.Runtime.DbgEngDataReader"), processDumpFile);
+        var debugInterface = reader.GetType().GetProperty("DebuggerInterface", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(reader);
+        var wrapper = new DataReaderWrapper(reader);
+        target = (DataTarget)typeof(DataTarget).GetMethod("CreateFromReader", BindingFlags.Static | BindingFlags.NonPublic).Invoke(null, new[] { wrapper, debugInterface });
```

Comment:
😟 Maybe there are new public APIs to avoid this reflection?

Reply:
On further review, it appears these changes were part of a separate investigation, and not required for this pull request.